nebula_hw_interfaces/src/nebula_hesai_hw_interfaces/hesai_hw_interface.cpp

```diff
@@ -2367,29 +2376,29 @@ Status HesaiHwInterface::SetPtpConfig(
 }
 Status HesaiHwInterface::SetPtpConfig(
   std::shared_ptr<boost::asio::io_context> ctx, int profile, int domain, int network,
-  int logAnnounceInterval = 1, int logSyncInterval = 1, int logMinDelayReqInterval = 0,
-  bool with_run)
+  int switch_type, int logAnnounceInterval = 1, int logSyncInterval = 1,
```

Comment:
The default parameters were already there before, but usually all parameters after ones that have defaults, have to have defaults too. We should probably add a default for `with_run` or get rid of all of them.
Also, I'm curious as to why the compiler doesn't give a warning..

Reply:
Oops. The reason is that there is no compiler error, is because the `with_run` default is in the header file. This is not very good, we should move all the defaults to the header file. Nice catch.

Reply:
Resolved in https://github.com/tier4/nebula/pull/125/commits/9a28e3493b452158792ae83bf171c9a47efdefe3